docs/source/en/guides/upload.md

```diff
@@ -103,6 +103,70 @@ set, files are uploaded at the root of the repo.
+- it is resumable: the upload process is split in many small tasks (hashing files, pre-uploading them and committing them). Each time a task is completed, the result is cached locally in a `./cache/huggingface` folder inside the folder you are trying to upload. By doing so, restarting the process after an interruption will resume all completed tasks.
+- it is multi-threaded: hashing large files and pre-uploading them benefits a lot from multithreading if your machine allows it.
+- it is resilient to errors: a high-level retry-mechanism has been added to retry each independent task indefinitely until it passes (no matter if it's a OSError, ConnectionError, PermissionError, etc.). This mechanism is double-edged. If transient errors happen, the process will continue and retry. If permanent errors happen (e.g. permission denied), it will retry indefinitely without solving the root cause.
+
+If you want more technical details about how `large_upload` is implemented under the hood, please have a look to the [`large_upload`] package reference.
+
+Here is how to use [`large_upload`] in a script. The method signature is very similar to [`upload_folder`]:
+
+```py
+>>> api.upload_folder(
+...     repo_id="HuggingFaceM4/Docmatix",
+...     repo_type="dataset",
+...     folder_path="/path/to/local/docmatix",
+... )
+```
+
+You will see the following output in your terminal:
+```
+Repo created: https://huggingface.co/datasets/HuggingFaceM4/Docmatix
+Found 5 candidate files to upload
+Recovering from metadata files: 100%|█████████████████████████████████████| 5/5 [00:00<00:00, 542.66it/s]
+
+---------- 2024-07-22 17:23:17 (0:00:00) ----------
+Files:   hashed 5/5 (5.0G/5.0G) | pre-uploaded: 0/5 (0.0/5.0G) | committed: 0/5 (0.0/5.0G) | ignored: 0
+Workers: hashing: 0 | get upload mode: 0 | pre-uploading: 5 | committing: 0 | waiting: 11
+---------------------------------------------------
+```
+
+First, the repo is created if it didn't exist before. Then the local folder is scanned for files to upload. For each file, we try to recover metadata information (from a previously interrupted upload). From there, it is able to launch workers and print an update status every 1 minute. Here we can see that 5 files have already been hashed but not pre-uploaded. 5 workers are pre-uploading files while the 11 others are waiting for a task.
+
+A command line is also provided. You can define the number of workers and the level of verbosity in the terminal:
+
+```sh
+huggingface-cli large-upload HuggingFaceM4/Docmatix --repo-type=dataset /path/to/local/docmatix --num-workers=16
+```
+
+<Tip>
+
+For large uploads, you have to set `repo_type="model"` or `--repo-type=model` explicitly. Usually this information is implicit in all other `HfApi` methods. This is to avoid having data uploaded to a repository with a wrong type. If that's the case, you'll have to re-upload everything.
```

Comment:
```suggestion
For large uploads, you have to set `repo_type="model"` or `--repo-type=model` explicitly. Usually, this information is implicit in all other `HfApi` methods. This is to avoid having data uploaded to a repository with a wrong type. If that's the case, you'll have to re-upload everything.
```